client/report_event_client.go

```diff
@@ -28,6 +28,13 @@ type KafkaProducer interface {
 
 type marshalFunc func(s interface{}) ([]byte, error)
 
+// ErrorReporter is the interface that wraps the ReportError method.
+// instanceID and errContent are required parameters. If neither is provided or there is any error while attemptin to
+// report the event then an error is retuned which the caller can handle as they see fit.
+type ErrorReporter interface {
+	ReportError(instanceID string, errContext string, err error, data log.Data) error
```

Comment:
Do we want to make this more generic than instanceID or will it only be for the import process?